deploy/rest/config.yaml

```diff
@@ -0,0 +1,231 @@
+components:
+  - model
+  - listener
+  - vector_index
+  - datatype
+
+leaves:
+  query:
+    MongoQuery:
+      _path: superduperdb/backends/mongodb/query/parse_query
+      query:
+        type: code
+      documents:
+        type: json
+        default: []
+  code:
+    Code:
+      _path: superduperdb/Code
+      identifier:
+        type: str
+      code:
+        type: str
+        default: |
+          from superduperdb import code
+
+          @code
+          def my_code(x):
+            return x
+  lazy_artifact:
+    LazyArtifact:
+      _path: superduperdb/components/datatype/LazyArtifact
+      identifier:
+        type: str
+      file_id:
+        type: blob
+  vector_index:
+    VectorIndex:
+      _path: superduperdb/VectorIndex
+      identifier:
+        type: str
+      measure:
+        type: str
+        choices:
+          - cosine
+          - dot
+          - l2
+      indexing_listener:
+        type: listener
+      compatible_listener:
+        type: listener
+        optional: True
+  datatype:
+    image:
+      _path: superduperdb/ext/pillow/image_type
+      identifier:
+        type: str
+      media_type:
+        type: str
+        default: image/png
+    vector:
+      _path: superduperdb/vector
+      identifier:
+        type: str
+      shape:
+        type: int
+  stack:
+    Stack:
+      _path: superduperdb/Stack
+      identifier:
+        type: str
+      components:
+        type: [model, listener, vector_index]
```

Comment:
type is used to define what kind of form is rendered. the type of type should be always `str`